src/components/withCurrentUserPersonalDetails.js

```diff
@@ -26,25 +26,28 @@ export default function (WrappedComponent) {
         /** Session of the current user */
         session: PropTypes.shape({
             email: PropTypes.string,
+            accountID: PropTypes.number,
         }),
     };
     const defaultProps = {
         forwardedRef: undefined,
         personalDetails: {},
         session: {
             email: '',
+            accountID: 0,
         },
     };
 
     const WithCurrentUserPersonalDetails = (props) => {
         const currentUserEmail = props.session.email;
-
+        const accountID = props.session.accountID;
+        const currentUserPersonalDetails = useMemo(() => ({...props.personalDetails[currentUserEmail], accountID}), [props.personalDetails, currentUserEmail, accountID]);
```

Comment:
I think it's okay to keep it, or it's even better to keep it. If we don't, the wrapped component may re-render on every update since the passed `currentUserPersonalDetails` object would be different, e.g. when the wrapped component is using `React.memo`, if we keep constructing a new `currentUserPersonalDetails` object every time, the `React.memo` will be useless.

Reply:
ah I see that makes sense!